source/common/common/logger.cc

```diff
@@ -55,29 +61,43 @@ void DelegatingLogSink::set_formatter(std::unique_ptr<spdlog::formatter> formatt
 
 void DelegatingLogSink::log(const spdlog::details::log_msg& msg) {
   absl::ReleasableMutexLock lock(&format_mutex_);
-  if (!formatter_) {
-    lock.Release();
-    sink_->log(absl::string_view(msg.payload.data(), msg.payload.size()));
-    return;
+  absl::string_view msg_view = absl::string_view(msg.payload.data(), msg.payload.size());
+  if (formatter_) {
+    fmt::memory_buffer formatted;
+    formatter_->format(msg, formatted);
+    msg_view = absl::string_view(formatted.data(), formatted.size());
   }
-
-  fmt::memory_buffer formatted;
-  formatter_->format(msg, formatted);
   lock.Release();
-  sink_->log(absl::string_view(formatted.data(), formatted.size()));
+
+  if (should_escape_) {
+    msg_view = absl::StripSuffix(msg_view, "\n");
+    msg_view = absl::StripSuffix(msg_view, "\r");
+    sink_->log(absl::CEscape(msg_view));
+    sink_->log("\r\n");
```

Comment:
+1 -- I don't think we want \r in the messages on non-windows platforms, do we?

Reply:
Yep, stripping is fine, but let's add back in the platform specific EOLs. Seems fine to use `spdlog`, but if there is STL or Abseil equivalents, those would be even nicer as we don't want to form super hard dependencies on `spdlog`.

Reply:
The spdlog version seems like the best option. Anyways, all it does is switch the eol character using `#ifdef _WIN32`. If we do need to switch in the future, we can copy that switch into our own utility function.

Reply:
Fair enough, we can always replace later.

Reply:
That is OK. Another option to consider is to keep the original whitespace suffix from the input string. Then there's a single source of truth about how to terminate lines, and this code doesn't need to be opinionated about it. WDYT?

Reply:
I could do it that way, but it just seemed a lot messier. I'd have to hard-code in checks for `\r\n` or just `\n`, and then add them back in after calling `sink_->log(absl::CEscape(msg_view));` (The checks have to be hardcoded because the EOL can be either 1 or 2 characters).